src/Microsoft.AspNetCore.Blazor/Routing/Router.cs

```diff
@@ -56,6 +46,8 @@ public void Init(RenderHandle renderHandle)
         public void SetParameters(ParameterCollection parameters)
         {
             parameters.AssignToProperties(this);
+            var types = ComponentResolver.ResolveComponents(AppAssembly);
+            Routes = RouteTable.Create(types);
```

Comment:
Instead of re-scanning the assemblies and building a new `RouteTable` for each `Router` instance or when `SetParameters` gets called, could we cache the `RouteTable` for any given `AppAssembly`? For example in a `static IDictionary<Assembly, RouteTable>`.

I know it would be uncommon for `Router.SetParameters` to be called more than once, but it is possible depending on how people structure their apps. Also when we get onto support for nested `Router` instances it will become common.

Reply:
That's fair. How do you feel about statics vs DI for framework components?

Reply:
I don't necessarily have a preference. What do you think would work better? We are using statics for caching stuff in other places already. For any case where we do use a `static`, consider SSR: it's important that it never leaks any state that would be different from one user to another. In this example I think that's OK because the computed route table would always be identical for any given `AppAssembly`.